src/libs/ajax.js

```diff
@@ -260,6 +260,18 @@ export const DAR = {
     return res.data;
   },
 
+  downloadDARDocument: async (referenceId, fileType, fileName) => {
```

Comment:
The method right above this one (line 252 `getDARDocument`), is what's called when we download the file from the DAR form. Currently, it uses the UUID of the file in GCS as the file name which is pretty poor UX. Can we either refactor that (see usages in `UploadLabelButton`) or file a ticket to do that?

Reply:
I'm glad you caught this.  I want to do a bigger refactor here and recommend we make a new ticket for that.  May I do that?

Wherever we're using "createObjectURL" and we're not using revokeObjectURL to remove it, we're potentially leaking memory.  See: https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL

This happens in `UploadLabelButton`, `downloadDataSets`, and in utils `download`

And to your point, I'd also like to make sure we're providing file names whenever possible.

Reply:
See [DUOS-2190](https://broadworkbench.atlassian.net/browse/DUOS-2190)

Reply:
> I'm glad you caught this. I want to do a bigger refactor here and recommend we make a new ticket for that. May I do that?

Absolutely. See also https://broadworkbench.atlassian.net/browse/DUOS-2188 where I'd like to do a much better job of handling DAR files.